app/src/main/java/de/danoeh/antennapod/fragment/FeedItemlistFragment.java

```diff
@@ -552,8 +578,11 @@ private void loadFeedImage() {
     private void loadItems() {
         if (disposable != null) {
             disposable.dispose();
+        } else {
+            //only show on first loading
+            progressBar.setVisibility(View.VISIBLE);
         }
-        progressBar.setVisibility(View.VISIBLE);
+
```

Comment:
I think I left this in on purpose because I saw users with database exports so big that it took around a second to mark something as played. If we don't display a progress bar in that case, the app feels sluggish. If this is handled with a `FeedItemEvent` instead, there should be no flickering.

Reply:
> I still don't really like the fact that you swipe away an episode and then it comes back with another state. Would it be possible to limit the distance that the item can be swiped? Maybe there is a way to make it feel similar to the pull-to-refresh feature (You swipe, it becomes visible but inactive. You swipe more, it becomes active. You swipe even more, it no longer moves). If the action is actually removing the item from the list (like in the inbox or maybe even when filtered), it should still be a full swipe.

Hmm, not that easily at least i believe, because ItemTouchHelper.SimpleCallback only has onSwiped which is triggered if the item is swiped out completly. If you know anything else please tell me.

Also i found this in quite a few other apps (swipe to change state) so i doesnt seem strange to me, also, depending on which screen you are on (and which filter you choose) the item does disappear (like if you have a mark as played swipe action, in unified episodes on the first quickfilter (all) the state is changed, on the second quickfilter (only unplayed) the item is removed if marked as played). So the swipe action would have to check beforehand wether the item would be removed or not, which seems quite messy with three screens (Inbox, Episodes, Feeds) with possibly filters applied and several swipeactions - so theres a huge number of possible combinations...

Reply:
See other comment, this should be the same action as from context menu...